### DISCUSSION TITLE
Warning about unused foreach variable

### DISCUSSION THREAD

**Discussion by NN---** [dotnet/roslyn#70643] [Q&A] [answered]
This code produces only message in Visual Studio: `IDE0059: Unnecessary assignment of a value to 'x'`

```cs
foreach (var x in args)
{
}
```

How can I ask compiler to generate a warning ?

**Comment by CyrusNajmabadi** [comment]
You're assigning the enumerated elements to that variable, but then not using the variable.  The analyzer is telling you that that's very strange and likely indicates a code bug.

**Reply by NN---** [reply]
This what I expect.
But there is no compilation error.

**Reply by RikkiGibson** [reply, marked as answer]
I think the diagnostic in the original post is a code style diagnostic. Please see https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/overview?tabs=net-7#enable-on-build.

**Reply by NN---** [reply]
Thanks.
I didn't set EnforceCodeStyleInBuild.
